src/main/java/com/epam/reportportal/service/ReportPortal.java

```diff
@@ -402,7 +402,7 @@ protected RestEndpoint buildRestEndpoint(ListenerParameters parameters, HttpClie
 					}},
 					new ReportPortalErrorHandler(jacksonSerializer),
 					buildEndpointUrl(baseUrl, project, parameters.isAsyncReporting()),
-					executorService
+					buildExecutorService(parameters)
```

Comment:
Well, it looks like I need to introduce breaking changes. The idea here was to use single executor service for everything, to be able to wait for its drain. Building new one will cause a situation when not all logs/items will be reported on client close.

Anyway, thanks for pointing that out. I'll try to fix it with a high priority in the next release.